src/calculatePlayerMoves.ts

```diff
@@ -0,0 +1,243 @@
+
+  // When square is already occupied by other player
+  if (square.playerId) return false;
+
+  // When there is a wall between player and targeted square
+  const coordinatesWallInBetween: Coordinates = {
+    y: coordinatesPlayer.y + (coordinatesSquare.y - coordinatesPlayer.y) / 2, // division by 2, because coordinate difference between player and targeted square is 2 and we select the middle square
+    x: coordinatesPlayer.x + (coordinatesSquare.x - coordinatesPlayer.x) / 2
+  };
+  const wallInBetween = getSquareByCoordinates(coordinatesWallInBetween, board);
+  if (wallInBetween.type !== SquareType.Wall) throw new Error("Square is not of type 'Wall'");
+  if (!wallInBetween.isWalkable) return false;
+  return true;
+};
+
+// Checks if a player can jump over to a square 2 blocks away in each direction (depends on which this is called).
+// In order to be able to, there have to be no blocking walls and there has to other player between targeted and current position squares.
+const checkLinearJump = (coordinatesSquare: Coordinates, coordinatesPlayer: Coordinates, board: Board): boolean => {
+  const square = getSquareByCoordinates(coordinatesSquare, board);
+  if (square.type !== SquareType.Player) throw new Error("Square is not of type 'Player'");
+
+  // When there is a wall between player and targeted square (space closest to player)
+  let coordinatesWallInBetween: Coordinates = {
+    y: coordinatesPlayer.y + (coordinatesSquare.y - coordinatesPlayer.y) / 4, // division by 4, because coordinate difference between player and targeted square is 4 and we select the square closest to the player
+    x: coordinatesPlayer.x + (coordinatesSquare.x - coordinatesPlayer.x) / 4
+  };
+  let wallInBetween = getSquareByCoordinates(coordinatesWallInBetween, board);
+  if (wallInBetween.type !== SquareType.Wall) throw new Error("Square is not of type 'Wall'");
+  if (!wallInBetween.isWalkable) return false;
+
+  // When there no other player between player and targeted square
+  const coordinatesPlayerInBetween: Coordinates = {
+    y: coordinatesPlayer.y + (coordinatesSquare.y - coordinatesPlayer.y) / 2, // division by 2, because coordinate difference between player and targeted square is 4 and we select the middle square (x / 4 * 2 = x / 2)
+    x: coordinatesPlayer.x + (coordinatesSquare.x - coordinatesPlayer.x) / 2
+  };
+  const playerInBetween = getSquareByCoordinates(coordinatesPlayerInBetween, board);
+  if (playerInBetween.type !== SquareType.Player) throw new Error("Square is not of type 'Player'");
+  if (!playerInBetween.playerId) return false;
+
+  // When there is a wall between player and targeted square (space closest to targeted square)
+  coordinatesWallInBetween = {
+    y: coordinatesPlayer.y + ((coordinatesSquare.y - coordinatesPlayer.y) / 4) * 3, // division by 4 and multiplication by 3, because coordinate difference between player and targeted square is 4 and we select the square between the middle square and targeted square

```

Comment:
These calculations are becoming kind of difficult to read. What I recommend is writing a function like 
`const maxValue = (value, maxValue) =>  value > maxValue ? maxValue : value`
use this for any operations that need division

Reply:
You can even have a function that calculates the coordinates. For example
`const getCoordinatesInBetween = (coordinatesPlayer, coordinatesSquare, maxValue)` . You would the `maxValue` for the `maxValue` function that I wrote above and you could reuse this everywhere. If you don't understand what I mean, I can elaborate further